pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -228,3 +232,165 @@ func TestIPInPodSubnets(t *testing.T) {
 	assert.Equal(t, false, c.Controller.IPInPodSubnets(net.ParseIP("10.10.10.10")))
 	assert.Equal(t, false, c.Controller.IPInPodSubnets(net.ParseIP("8.8.8.8")))
 }
+
+func TestRemoveStaleTunnelPorts(t *testing.T) {
+	c, closeFn := newController(t, &config.NetworkConfig{
+		TrafficEncapMode:  0,
+		TunnelType:        ovsconfig.TunnelType("vxlan"),
+		EnableIPSecTunnel: true,
+		IPSecPSK:          "changeme",
+	})
+
+	c.interfaceStore.AddInterface(&interfacestore.InterfaceConfig{
+		Type:          interfacestore.TunnelInterface,
+		InterfaceName: util.GenerateNodeTunnelInterfaceName("xyz-k8s-0-1"),
+		TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+			NodeName: "xyz-k8s-0-1",
+			Type:     ovsconfig.TunnelType("vxlan"),
+			PSK:      "mismatchpsk",
+			RemoteIP: nodeIP1,
+		},
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "123",
+		},
+	})
+
+	defer closeFn()
+	defer c.queue.ShutDown()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
```

Comment:
maybe unify this set up code in a separate function:

```go
func setup(ifaces ...*interfacestore.InterfaceConfig) cleanupFn {
    ...
}
```

then in the test:
```go
cleanup := setup(ifaceConfig)
defer cleanup()
```

Reply:
done